### DISCUSSION TITLE
The Development Workflow

### DISCUSSION THREAD

**Discussion by barelyhuman** [standardnotes/mobile#427] [Q&A] [answered]
So just like you have the whole `dev` and `prod` variants setup for both android and ios, I wanted to know what other decisions where taken to make these seamless, I'm kinda working on a similar setup and really like the whole setup you have. 

My fastlane setup is on the simpler side because the use case doesn't involve the same amount of complexity and while most of my questions were answered during a quick browse on the code, I'd like someone to help me out as to what else was being handled in case I miss something. 

1. Setup dev and prod flavours in Android 
2. Setup dev and prod targets in iOS (separate icons for both types and separate bundle ids)
3. Add cert handling with fastfile for android and ios with match handling it with git 
4.  The same separation is handled in the appfile for the different variants

Then there's the other source code related stuff which I didn't go through in detail but assuming that there's no other environment based values in there and you just use the different variants and targets to separate concerns. 

Overall, is the above enough to setup the variants or is there anything else that needs to handled before I go with this approach?

Thanks!

**Comment by antsgar** [comment, marked as answer]
Hi @barelyhuman! Hope you're doing well 🙂 I think you've covered pretty much all of what needs to be done.

Specifically for Android, you can notice that you need to add the `productFlavours` to your `app/build.gradle` file https://github.com/standardnotes/mobile/blob/develop/android/app/build.gradle#L182, and add each flavour's icons to the corresponding res folder.

On iOS, you should create separate targets and schemes on XCode and add the targets to your `Podfile`, like so https://github.com/standardnotes/mobile/blob/develop/ios/Podfile#L16. You can also create separate icon sets and set them to each target on **Build Settings** > **Asset Catalog Compiler** - **Options** > **Asset Catalog App Icon Set Name**.

The rest is handled on the Fastlane files, like you mentioned. There are also separate scripts on `package.json` to run each flavour of the app on each OS.

Hope this helps a bit, and if you have any questions don't hesitate to reach out ☺️

**Reply by barelyhuman** [reply]
Yeah I ended up experimenting with everything I mentioned and it worked flawlessly. 
Thanks for the pointers again though @antsgar 

There's a good amount of difference in terms of the dev builds so the Fastfile wasn't used since I already had one but the iOS auto signing is something I wanted to work but well, manual is the only way that'll work. 
Anyway, Thanks again!
